src/app/static/src/tests/components/runWorkflow/runWorkflowCreate.test.ts

```diff
@@ -28,18 +93,118 @@ describe(`runWorkflowCreate`, () => {
+            expect(wrapper.vm.$data.error).toStrictEqual("")
+            expect(wrapper.vm.$data.defaultMessage).toStrictEqual("")
+            expect(wrapper.vm.$data.workflows).toStrictEqual(workflowSummaryMetadata)
+
+            const vueSelect = wrapper.find(VueSelect)
+            vueSelect.vm.$emit("input", selectedWorkflow)
+            await wrapper.setData({runWorkflowMetadata: workflowMetadata})
+
+            await wrapper.find("#rerun").trigger("click")
+            expect(wrapper.emitted("rerun").length).toBe(1)
+            done()
+        })
     })
 
-    it(`can emit clone navigation step`, async() => {
+    it(`can emit re-run metadata`, async (done) => {
         const wrapper = getWrapper()
-        expect(wrapper.find("h2").text()).toBe("Run workflow")
-        await wrapper.find("#clone").trigger("click")
-        expect(wrapper.emitted("clone").length).toBe(1)
+
+        setTimeout(async () => {
+            expect(wrapper.find("h2").text()).toBe("Run workflow")
+            expect(mockAxios.history.get.length).toBe(1)
+            const vueSelect = wrapper.find(VueSelect)
+            vueSelect.vm.$emit("input", selectedWorkflow)
+            await wrapper.setData({runWorkflowMetadata: workflowMetadata})
+
+            await wrapper.vm.$emit("rerun", runnableWorkflowMetadata)
+            expect(wrapper.emitted("rerun").length).toBe(1)
+            expect(wrapper.emitted().rerun[0]).toEqual([runnableWorkflowMetadata])
+            done()
+        })
     })
 
+    it(`can emit clone navigation step`, async (done) => {
+        const wrapper = getWrapper()
+
+        setTimeout(async () => {
+            expect(wrapper.find("h2").text()).toBe("Run workflow")
+            expect(mockAxios.history.get.length).toBe(1)
+            expect(mockAxios.history.get[0].url).toBe("http://app/workflows")
+            expect(wrapper.vm.$data.error).toStrictEqual("")
+            expect(wrapper.vm.$data.defaultMessage).toStrictEqual("")
+            expect(wrapper.vm.$data.workflows).toStrictEqual(workflowSummaryMetadata)
+
+            const vueSelect = wrapper.find(VueSelect)
+            vueSelect.vm.$emit("input", selectedWorkflow)
+            await wrapper.setData({runWorkflowMetadata: workflowMetadata})
+
+            await wrapper.find("#clone").trigger("click")
+            expect(wrapper.emitted("clone").length).toBe(1)
+            done()
+        })
+    })
+
+    it(`can emit clone metadata`, async (done) => {
+        const wrapper = getWrapper()
+
+        setTimeout(async () => {
+            expect(wrapper.find("h2").text()).toBe("Run workflow")
+            expect(mockAxios.history.get.length).toBe(1)
+            const vueSelect = wrapper.find(VueSelect)
+            vueSelect.vm.$emit("input", selectedWorkflow)
+            await wrapper.setData({runWorkflowMetadata: workflowMetadata})
+            await wrapper.find("#clone").trigger("click")
+            expect(wrapper.emitted("clone").length).toBe(1)
```

Comment:
After making the suggested change on line 163 you should check the actual contents of the clone event are as expected here. And ensure that similar changes are made to other tests in this file that use `workflowMetadata`.